src/lib/lore.sh

```diff
@@ -458,3 +458,41 @@ function download_series()
   done
   wait
 }
+
+function apply_patch()
+{
+  local total_patches="$1"
+  local first_message_id="$2"
+  local patch_title="$3"
+  local kernel_tree="$4"
+  local target_branch="$5"
+  local flag="$6"
+  local path_to_tmp_dir
+  local filename_pattern
+  local files_to_remove
+  local cmd
+  local ret
+
+  path_to_tmp_dir=$(mktemp --directory)
+
+  download_series "${total_patches}" "${patch_url}" "${path_to_tmp_dir}" "${patch_title}"
+
+  cmd="git -C ${kernel_tree} switch ${target_branch}"
+  cmd_manager "$flag" "$cmd"
+  ret="$?"
+  if [[ "$ret" != 0 ]]; then
+    complain "Could not switch kernel tree ${kernel_tree} to branch ${target_branch}"
+    return "$ret"
+  fi
+
+  filename_pattern=$(convert_title_to_patch_name "${patch_title}")
+  files_to_apply=$(find "${path_to_tmp_dir}" -iname "*${filename_pattern}" | sort)
```

Comment:
The default behavior of `sort` may vary across distros, using an explicit order, e.g.; `--dictionary-order`.